realm/src/main/java/io/realm/android/RealmCursor.java

```diff
@@ -0,0 +1,548 @@
+            } else {
+                result.getChars(0, result.length(), data, 0);
+            }
+            buffer.sizeCopied = result.length();
+        } else {
+            buffer.sizeCopied = 0;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public short getShort(int columnIndex) {
+        return (short) mapRealmTypeToCursor(columnIndex, false);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getInt(int columnIndex) {
+        return (int) mapRealmTypeToCursor(columnIndex, false);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getLong(int columnIndex) {
+        return mapRealmTypeToCursor(columnIndex, true);
+    }
+
+    private long mapRealmTypeToCursor(int columnIndex, boolean acceptLong) {
+        ColumnType type = table.getColumnType(columnIndex);
+        switch(type) {
+            case BOOLEAN: return table.getBoolean(columnIndex, rowIndex) ? 1 : 0;
+            case INTEGER: return table.getLong(columnIndex, rowIndex);
+            case DATE:
+                if (!acceptLong) {
+                    throw new IllegalArgumentException("Use getLong() instead to get date values.");
+                } else {
+                    return table.getDate(columnIndex, rowIndex).getTime();
+                }
+            default:
+                throw new IllegalArgumentException(String.format("Column at index  %s is not a Boolean or an Integer but a %s",
+                        columnIndex, type));
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public float getFloat(int columnIndex) {
+        return table.getFloat(columnIndex, rowIndex);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public double getDouble(int columnIndex) {
+        return table.getDouble(columnIndex, rowIndex);
+    }
+
+    /**
+     * Returns the SQLite Cursor type for a column index. Realm has more types than SQLite and some of them cannot be
+     * mapped to any meaningful SQLite type. These will return -1 and are cannot be accessed using a cursor.
+     * <p>
+     * Some realm types are mapped to conform to the Cursor interface. These mappings are described below:
+     * <ol>
+     *   <li>Boolean : {0 = false, 1 = true}, use getShort()/getInt()/getLong()</li>
+     *   <li>Date : Time in milliseconds since epoch, use getLong()</li>
+     *</ol>
+     *
+     * @param columnIndex Get the data type from the this column index.
+     * @return One of the {@code FIELD_TYPE_*}'s described in {@link Cursor}
+     */
+    @Override
+    public int getType(int columnIndex) {
+        ColumnType realmType = table.getColumnType(columnIndex);
+        switch (realmType) {
+            case BOOLEAN: return Cursor.FIELD_TYPE_INTEGER;
+            case INTEGER: return Cursor.FIELD_TYPE_INTEGER;
+            case FLOAT: return Cursor.FIELD_TYPE_FLOAT;
+            case DOUBLE: return Cursor.FIELD_TYPE_FLOAT;
```

Comment:
Not really since the cursor still have Cursor.getFloat() and Cursor.getDouble(), but SQLite doesn't have a way to expose the difference. So ultimately it is a developer responsibility :(